add GOVERNANCE.md (#4269)
cri-o/cri-o: GOVERNANCE.md

> # CRI-O contributor tiers

Reviewer: Let's link the file from the README.md with a short sentence describing its intention.

Reviewer 2: Good thought @saschagrunert I'd also suggest a link and a bit of explanatory verbiage on CONTRIBUTING.md

## Maintainer responsibilities

* Monitor our channel (#crio) on the Kubernetes Slack instance and our IRC channel (#cri-o) on Freenode (delayed response is perfectly acceptable)
* Triage GitHub issues and perform pull request reviews for other maintainers and the community. The areas of specialization listed in [OWNERS.md](OWNERS.md) can be used to help with routing an issue/question to the right person.
* Triage CI issues - file issues for known flaky CI runs or bugs, and either fix or find someone to fix any breakages.
* During GitHub issue triage, apply all applicable [labels](https://github.com/cri-o/cri-o/labels) to each new issue. Labels are extremely useful for future issue follow up. Which labels to apply is somewhat subjective so just use your best judgment. A few of the most important labels that are not self explanatory are:
  * **good first issue**: Mark any issue that can reasonably be accomplished by a new contributor with this label.
  * **help wanted**: Unless it is immediately obvious that someone is going to work on an issue (and if so assign it), mark it help wanted.

>   * **CRI change**: If sufficiently fixing the issue involves a change to the CRI, this label is given. Often, maintainers won't have time to make such a change, but the change could be seen as good and as such the issue isn't closed.

Reviewer 2: Do you have a 'kind/feature' label and do you want to also highlight that?

Author: that's more for PRs, for issues we have `area/vm` but I don't think it's common enough to warrant adding right now

## When does a maintainer lose maintainer status

If a maintainer is no longer interested or cannot perform the maintainer duties listed above, they may volunteer to be removed from the [OWNERS list](github.com/cri-o/cri-o/tree/master/OWNERS).

In extreme cases this can also occur by a vote of the maintainers per the voting process below.

They may also opt to be come an [emeritus maintainer](#emeritus-maintainer)

# Emeritus Maintainer

An emeritus maintainer is a maintainer who used to spend a significant amount of time on the project, but now has refocused their efforts.

The CRI-O community has the emeritus maintainer tier to recognize the impact and input early maintainers have had on the project, while also being realistic about their current committment level.

There is no formal process for becoming an emeritus maintainer: a maintainer simply slows down on their contributions but does not remove themselves from the [OWNERS file](OWNERS.md).

# Feature addition policy

Features are added by simply adding a PR. The maintainers handle this the same as a bug fix PR, discussing whether the feature will be added in the PR page.

# Conflict resolution and voting

> In general, we prefer that technical issues and maintainer membership are amicably worked out between the persons involved. If a dispute cannot be decided independently, the maintainers can be called in to decide an issue. If the maintainers themselves cannot decide an issue, the issue will be resolved by voting. The voting process is a simple majority in which each maintainer receives one vote. Votes not received within the span of one week are treated as an abstention.

Reviewer 2: Any tie breaker procedures?  If you list them, you'll never need them, if you don't....
Perhaps just "In the case of a tie vote, the issue will be considered to have not passed."

# CRI-O GOVERNANCE.md

> CRI-O, being a project whose purpose is linked to fulfilling the needs of Kubernetes, has a governance structure that mirrors many SIGs and sub-projects of Kubernetes. This document highlights important aspects of, and serves as a summary of the [Kubernetes' community documentation](https://www.kubernetes.dev/docs.

Reviewer: ```suggestion
CRI-O, being a project whose purpose is linked to fulfilling the needs of Kubernetes, has a governance structure that mirrors many SIGs and sub-projects of Kubernetes. This document highlights important aspects of, and serves as a summary of the [Kubernetes' community documentation](https://www.kubernetes.dev/docs).
```

Author: good catch!

# CRI-O contributor tiers

* CRI-O has four tiers of code contributors: approvers, reviewers, emeritus_approvers and community members.
* Anyone can (and is encouraged to) be a community member, whose responsibilities are up to the contributor (submitting PRs, reviewing PRs).
* Reviewers are a set of people responsible for reviewing PRs and triaging issues. In most cases, they're also responsible for fixing issues in the code base.
* Approvers are the set of people that have final say over the direction of the project. For any PR to merge, an approver must approve it.
* Emeritus Approvers can be considered "retired" maintainers--while they don't have approval rights, they are listed in the OWNERS file to indicate their expertise in the area.

## Community Member

* Anyone can and is encouraged to be a community member! The CRI-O community eagerly accept PRs, issues, new users and questions.

## Reviewer

### How to become a reviewer

* If you'd like to have a more active role in the community, and one day hope to become an approver, you may become a reviewer to build trust and deepen your knowledge of the code base. 

> * To become a reviewer, an you may file an issue with [this template](https://github.com/cri-o/cri-o/issues/new?assignees=&labels=area%2Fgithub-membership&template=membership.yml&title=REQUEST%3A+New+membership+for+%3Cyour-GH-handle%3E).

Reviewer 3: ```suggestion
* To become a reviewer, you may file an issue with [this template](https://github.com/cri-o/cri-o/issues/new?assignees=&labels=area%2Fgithub-membership&template=membership.yml&title=REQUEST%3A+New+membership+for+%3Cyour-GH-handle%3E).
```

* To find a sponsor, please feel free to reach out to the approvers. The approver team is happy to foster growth within the CRI-O community.
* 
### Reviewer responsibilities

* Once you are a reviewer, the CRI-O community will expect you to start contributing PRs, taking issues and doing code reviews under the guidance of current approvers.

> * Monitor our channel (#crio) on the Kubernetes Slack instance and answer questions where possible (delayed response is perfectly acceptable).

Reviewer: Let's add a link to the slack channel

  * **good first issue**: Mark any issue that can reasonably be accomplished by a new contributor with this label.
  * **help wanted**: Unless it is immediately obvious that someone is going to work on an issue (and if so assign it), mark it help wanted.
  * **CRI change**: If sufficiently fixing the issue involves a change to the CRI, this label is given. Often, maintainers won't have time to make such a change, but the change could be seen as good and as such the issue isn't closed.
* Make sure that ongoing PRs are moving forward at the right pace or closing them.

## Approver
### How to become an approver

* The barrier to become an approver is naturally higher than that of becoming a reviewer
* There isn't an explicitly spelled out set of criteria that is required, but generally being an approver means you will be spending significant time maintaining and reviewing CRI-O code. 
* To become an approver, a simple majority vote among the current approvers should be performed. See [conflict resolution](#conflict-resolution-and-voting) for more information on how to handle disputes.

### Approver responsibilities

> * All of the responsibilities of the reviewers, and...

Reviewer: ```suggestion
Approvers have all of the responsibilities of the reviewers, as well as:
```

## When does a maintainer lose maintainer status

If a maintainer is no longer interested or cannot perform the maintainer duties listed above, they may volunteer to be removed from the [OWNERS list](github.com/cri-o/cri-o/tree/master/OWNERS).

In extreme cases this can also occur by a vote of the maintainers per the voting process below.

They may also opt to be come an [emeritus maintainer](#emeritus-maintainer)

# Emeritus Maintainer

An emeritus maintainer is a maintainer who used to spend a significant amount of time on the project, but now has refocused their efforts.

The CRI-O community has the emeritus maintainer tier to recognize the impact and input early maintainers have had on the project, while also being realistic about their current committment level.

> There is no formal process for becoming an emeritus maintainer: a maintainer simply slows down on their contributions but does not remove themselves from the [OWNERS file](OWNERS.md).

Reviewer 4: @haircommander, I really like the "Emeritus" idea, GNOME dos something quite similar.

If I may make a suggestion, I'd prefer having the Emeritus Maintainers listed as so. By doing this new contributors will know who they can ping, who they can ask for help, etc.  With no clear distinction between both may be a slightly confusing for new contributors (I must admit it was slightly confusing for me when I first arrived here a few months ago).
